### ISSUE TITLE
Error when attaching files to email notification

### ISSUE THREAD

**Issue by michaelhue** [barrelstrength/craft-sprout-forms#200]
### Description

Submitting a form with email notifications including attachments results in an exception:

```
Swift_IoException: Unable to open file for reading [/var/www/virtual/mlnds/app/storage/volumes/uploadsfpmv2019/test-projekt//Vorlage-Kostenfinanzierungsplan-Forderpreis-Musikvermittlung_181130_163552.xlsx] in /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/ByteStream/FileByteStream.php:131
Stack trace:
#0 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/ByteStream/FileByteStream.php(77): Swift_ByteStream_FileByteStream->getReadHandle()
#1 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/ContentEncoder/Base64ContentEncoder.php(40): Swift_ByteStream_FileByteStream->read(8192)
#2 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/SimpleMimeEntity.php(555): Swift_Mime_ContentEncoder_Base64ContentEncoder->encodeByteStream(Object(Swift_ByteStream_FileByteStream), Object(Swift_Transport_StreamBuffer), 0, 76)
#3 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/SimpleMimeEntity.php(532): Swift_Mime_SimpleMimeEntity->bodyToByteStream(Object(Swift_Transport_StreamBuffer))
#4 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/SimpleMimeEntity.php(570): Swift_Mime_SimpleMimeEntity->toByteStream(Object(Swift_Transport_StreamBuffer))
#5 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/SimpleMimeEntity.php(532): Swift_Mime_SimpleMimeEntity->bodyToByteStream(Object(Swift_Transport_StreamBuffer))
#6 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mime/SimpleMessage.php(604): Swift_Mime_SimpleMimeEntity->toByteStream(Object(Swift_Transport_StreamBuffer))
#7 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Message.php(162): Swift_Mime_SimpleMessage->toByteStream(Object(Swift_Transport_StreamBuffer))
#8 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(398): Swift_Message->toByteStream(Object(Swift_Transport_StreamBuffer))
#9 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(500): Swift_Transport_AbstractSmtpTransport->streamMessage(Object(Swift_Message))
#10 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(516): Swift_Transport_AbstractSmtpTransport->doMailTransaction(Object(Swift_Message), 'mlnds@uber.spac...', Array, Array)
#11 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/AbstractSmtpTransport.php(206): Swift_Transport_AbstractSmtpTransport->sendTo(Object(Swift_Message), 'mlnds@uber.spac...', Array, Array)
#12 /var/www/virtual/mlnds/app/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Mailer.php(71): Swift_Transport_AbstractSmtpTransport->send(Object(Swift_Message), Array)
#13 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2-swiftmailer/src/Mailer.php(156): Swift_Mailer->send(Object(Swift_Message))
#14 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/mail/BaseMailer.php(263): yii\swiftmailer\Mailer->sendMessage(Object(craft\mail\Message))
#15 /var/www/virtual/mlnds/app/vendor/craftcms/cms/src/mail/Mailer.php(175): yii\mail\BaseMailer->send(Object(craft\mail\Message))
#16 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-base/src/app/email/mailers/DefaultMailer.php(170): craft\mail\Mailer->send(Object(craft\mail\Message))
#17 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-base/src/app/email/services/NotificationEmails.php(118): barrelstrength\sproutbase\app\email\mailers\DefaultMailer->sendNotificationEmail(Object(barrelstrength\sproutbase\app\email\elements\NotificationEmail))
#18 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-base/src/app/email/services/NotificationEmailEvents.php(199): barrelstrength\sproutbase\app\email\services\NotificationEmails->sendNotificationViaMailer(Object(barrelstrength\sproutbase\app\email\elements\NotificationEmail))
#19 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-base/src/app/email/services/NotificationEmailEvents.php(154): barrelstrength\sproutbase\app\email\services\NotificationEmailEvents->handleDynamicEvent('barrelstrength\\...', Object(barrelstrength\sproutforms\events\OnSaveEntryEvent), Object(barrelstrength\sproutforms\integrations\sproutemail\events\notificationevents\SaveEntryEvent))
#20 [internal function]: barrelstrength\sproutbase\app\email\services\NotificationEmailEvents->barrelstrength\sproutbase\app\email\services\{closure}('barrelstrength\\...', Object(barrelstrength\sproutforms\events\OnSaveEntryEvent), Object(barrelstrength\sproutforms\integrations\sproutemail\events\notificationevents\SaveEntryEvent))
#21 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-base/src/app/email/services/NotificationEmailEvents.php(124): call_user_func(Object(Closure), 'barrelstrength\\...', Object(barrelstrength\sproutforms\events\OnSaveEntryEvent), Object(barrelstrength\sproutforms\integrations\sproutemail\events\notificationevents\SaveEntryEvent))
#22 [internal function]: barrelstrength\sproutbase\app\email\services\NotificationEmailEvents->barrelstrength\sproutbase\app\email\services\{closure}(Object(barrelstrength\sproutforms\events\OnSaveEntryEvent))
#23 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/Event.php(310): call_user_func(Object(Closure), Object(barrelstrength\sproutforms\events\OnSaveEntryEvent))
#24 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/Component.php(636): yii\base\Event::trigger('barrelstrength\\...', 'afterSave', Object(barrelstrength\sproutforms\events\OnSaveEntryEvent))
#25 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-forms/src/services/Entries.php(505): yii\base\Component->trigger('afterSave', Object(barrelstrength\sproutforms\events\OnSaveEntryEvent))
#26 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-forms/src/services/Entries.php(317): barrelstrength\sproutforms\services\Entries->callOnSaveEntryEvent(Object(barrelstrength\sproutforms\elements\Entry), true)
#27 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-forms/src/controllers/EntriesController.php(153): barrelstrength\sproutforms\services\Entries->saveEntry(Object(barrelstrength\sproutforms\elements\Entry))
#28 /var/www/virtual/mlnds/app/vendor/barrelstrength/sprout-forms/src/controllers/EntriesController.php(133): barrelstrength\sproutforms\controllers\EntriesController->saveEntryInCraft(Object(barrelstrength\sproutforms\elements\Entry))
#29 [internal function]: barrelstrength\sproutforms\controllers\EntriesController->actionSaveEntry()
#30 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#31 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#32 /var/www/virtual/mlnds/app/vendor/craftcms/cms/src/web/Controller.php(104): yii\base\Controller->runAction('save-entry', Array)
#33 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-entry', Array)
#34 /var/www/virtual/mlnds/app/vendor/craftcms/cms/src/web/Application.php(282): yii\base\Module->runAction('sprout-forms/en...', Array)
#35 /var/www/virtual/mlnds/app/vendor/craftcms/cms/src/web/Application.php(542): craft\web\Application->runAction('sprout-forms/en...', Array)
#36 /var/www/virtual/mlnds/app/vendor/craftcms/cms/src/web/Application.php(266): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#37 /var/www/virtual/mlnds/app/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#38 /var/www/virtual/mlnds/app/web/index.php(21): yii\base\Application->run()
#39 {main}
```

It seems the file path is not created correctly. The path is shown as:
`/var/www/virtual/mlnds/app/storage/volumes/uploadsfpmv2019/test-projekt//Vorlage-Kostenfinanzierungsplan-Forderpreis-Musikvermittlung_181130_163552.xlsx`

The correct path would be:
`/var/www/virtual/mlnds/app/storage/volumes/uploads/fpmv2019/test-projekt/Vorlage-Kostenfinanzierungsplan-Forderpreis-Musikvermittlung_181130_163552.xlsx`

### Steps to reproduce

1. Create a form with an upload field.
2. Create an email notification and set the "Attach Files" toggle to ON.
3. Submit the form.

### Additional info

- Craft version: Craft Pro 3.0.33
- PHP version: 7.1.24
- Database driver & version: MySQL 5.5.5
- Plugins & versions: 3.0.0-beta.34

**Comment by michaelhue** [comment]
The problem is `barrelstrength\sproutbase\app\email\mailers\DefaultMailer::getAssetFilePath()` which creates nonsensical paths.

I suggest the following implementation:

```
protected function getAssetFilePath(Asset $asset)
{
        $path = $asset->getVolume()->getRootPath() . DIRECTORY_SEPARATOR . $asset->getPath();

        return FileHelper::normalizePath($path);
}
```

Let me know if you'd like a pull request for that change.

**Comment by andrelopez** [comment]
Issue moved to [barrelstrength/craft-sprout-email #88](https://github.com/barrelstrength/craft-sprout-email/issues/88) via [**ZenHub**](https://www.zenhub.com/)

**Comment by BenParizek** [comment]
@michaelhue This issue should be resolved in v4.0.0-beta.9
